Automattic/wp-calypso
# InlineHelp: Record event when clicking on support pages in customer home

#### Changes proposed in this Pull Request

* When using the inline help system on the "My Home" page, begin sending tracking events when answers are clicked.
  * Tracking events are currently sent through help articles accessed via the FAB, the question mark button at the bottom of some pages;  However, no events are sent when the same articles are accessed from the "My Home" search. This unifies the events. An attribute specifies if the event was sent from FAB or "My Home". 

#### Testing instructions

* **Prerequisites**
  * Check out this branch and run calypso.
  * Visit http://calypso.localhost:3000/ and open the debug console.
  * Type `localStorage.debug = 'calypso:analytics*';`, press enter, then refresh the page
* **Test New Method** _(Tracking added by this PR)_
  * Visit "My Home" in the left sidebar (url like `/home/[domain]`)
  * Scroll down to the "Get Help" heading
  * Type "stats" in the "Search support articles" box
  * Results should appear
  * Click the support article named "stats", which links to "https://wordpress.com/support/stats/"
  * A tracking event is now sent (Previously, none was sent)
    * Note:  You will see two lines in the JS console for each event sent.  "Record Event" and "Recording Event".  This is expected.  There is only one event.

![2020-08-11_13-23](https://user-images.githubusercontent.com/937354/89934352-10d37800-dbd6-11ea-9fe6-9eb08ee1ed0f.png)

![2020-08-11_13-24](https://user-images.githubusercontent.com/937354/89934362-1466ff00-dbd6-11ea-858b-c72b207a81e2.png)

* **Compare with Old Method** _(Tracking already on production)_
  * Visit "Stats" in the left sidebar (url like `/stats/day/[domain]`)
  * Click the FAB (Question mark in bottom right)
  * Type "stats"
  * Results should appear
  * Click the support article named "stats", which links to "https://wordpress.com/support/stats/"
  * See a pop up with a preview.  Click the pink button labeled "Read More".
  * A tracking event is sent

![2020-08-11_13-26](https://user-images.githubusercontent.com/937354/89934502-4a0be800-dbd6-11ea-92f1-9f96260c2f88.png)
![2020-08-11_13-28](https://user-images.githubusercontent.com/937354/89934672-8dfeed00-dbd6-11ea-85b7-328ecf3f521d.png)



Worried about tours or video?  From my research, they don't apply here.  They cannot be found on the "My Home" "Get Help" search box.  They only appear on some pages when clicking the FAB (the question mark at the bottom of some pages).  For example, visiting /me/account, clicking the FAB, entering no text and clicking "Change your password" links to a video, but searching "Change your password" on the "My Home" "Get Help" search box does not.  [Code link](https://github.com/Automattic/wp-calypso/blob/057729962d6163aca68cab866e088c507365b464/client/blocks/inline-help/contextual-help.js#L1026)

Fixes https://github.com/Automattic/wp-calypso/issues/44027
